### PR TITLE
Fix for policy manager not setting labels in some cases

### LINKED ISSUE #511: Global policy got deleted even though I didn't do it
Background: I started my repo with `jarip@desktop-16aq323`, then I figured I have an ugly hostname and username, started a new backup with by changing the username and hostname to `jari@jari-desktop`, snapshotted again, and started to remove old snapshots/policies. 

Following happened:

Global policy got deleted, even though I didn't do it.

Following transcript:

```shell
PS C:\Users\jarip> kopia policy list
248e14f53cef48dbe6a9f8fe082d773b (global)
af787532bf6f08b24e14088eab529215 jari@jari-desktop:C:\Source
6d87cee5f814aaea6c304f6e4c6af88f jari@jari-desktop:C:\Zotero
e4b8a8ad181acc8c2632e8f3c6b5b2a6 jarip@desktop-16aq323:C:\Source
254b201bbbd115198f53be8ee065fd11 jarip@desktop-16aq323:C:\Zotero

PS C:\Users\jarip> kopia policy remove e4b8a8ad181acc8c2632e8f3c6b5b2a6
01:11:37.870 [kopia/cli] Removing policy on "(global)"...

PS C:\Users\jarip> kopia policy remove 254b201bbbd115198f53be8ee065fd11
01:11:45.870 [kopia/cli] Removing policy on "(global)"...

PS C:\Users\jarip> kopia policy list
af787532bf6f08b24e14088eab529215 jari@jari-desktop:C:\Source
6d87cee5f814aaea6c304f6e4c6af88f jari@jari-desktop:C:\Zotero
```

P.S. I was trying to remove old host (snapshots and policies).

### PR DESCRIPTION
**Pull request by jkowalski**
Fixed #511

### CODE REVIEW

File: snapshot/policy/policy_manager_test.go [hunk 1]
```diff
@@ -2,12 +2,163 @@ package policy
...
+		{
+			sourceInfo:    GlobalPolicySourceInfo,
+			wantEffective: defaultPolicyWithLabels,
+		},
+		{
+			sourceInfo: snapshot.SourceInfo{
+				UserName: "myuser",
+				Host:     "host-c",
+				Path:     "/some/path",
+			}, wantEffective: policyWithLabels(DefaultPolicy, map[string]string{
```
Comment (julio-lopez) [inline snapshot/policy/policy_manager_test.go:67]: nit: same below
```suggestion
			},
			wantEffective: policyWithLabels(DefaultPolicy, map[string]string{
```

File: snapshot/policy/policy_manager_test.go [hunk 2]
```diff
@@ -2,12 +2,163 @@ package policy
...
+				"path":       "/some/path",
+				"username":   "myuser",
+			}),
+		},
+		{
+			sourceInfo: snapshot.SourceInfo{
+				UserName: "myuser",
+				Host:     "host-a",
+				Path:     "/some/path",
+			}, wantEffective: policyWithLabels(defaultPolicyWithKeepDaily(t, 44), map[string]string{
```
Comment (julio-lopez) [inline snapshot/policy/policy_manager_test.go:80]:
```suggestion
			},
			wantEffective: policyWithLabels(defaultPolicyWithKeepDaily(t, 44), map[string]string{
```

File: snapshot/policy/policy_manager_test.go [hunk 3]
```diff
@@ -2,12 +2,163 @@ package policy
...
+				"type":       "policy",
+				"policyType": "path",
+				"hostname":   "host-a",
+				"path":       "/some/path",
+				"username":   "myuser",
+			}), wantSources: []snapshot.SourceInfo{
```
Comment (julio-lopez) [inline snapshot/policy/policy_manager_test.go:86]:
```suggestion
			}),
			wantSources: []snapshot.SourceInfo{
```

File: snapshot/policy/policy_manager_test.go [hunk 4]
```diff
@@ -2,12 +2,163 @@ package policy
...
+				{Host: "host-a"},
+			},
+		},
+		{
+			sourceInfo: snapshot.SourceInfo{
+				UserName: "myuser",
+				Host:     "host-a",
+				Path:     "/some/path2/nested",
+			}, wantEffective: policyWithLabels(defaultPolicyWithKeepDaily(t, 66), map[string]string{
```
Comment (julio-lopez) [inline snapshot/policy/policy_manager_test.go:95]:
```suggestion
			},
			wantEffective: policyWithLabels(defaultPolicyWithKeepDaily(t, 66), map[string]string{
```

File: snapshot/policy/policy_manager_test.go [hunk 5]
```diff
@@ -2,12 +2,163 @@ package policy
...
+				"type":       "policy",
+				"policyType": "path",
+				"hostname":   "host-a",
+				"path":       "/some/path2/nested",
+				"username":   "myuser",
+			}), wantSources: []snapshot.SourceInfo{
```
Comment (julio-lopez) [inline snapshot/policy/policy_manager_test.go:101]:
```suggestion
			}),
			wantSources: []snapshot.SourceInfo{
```

Comment (jkowalski) [reply snapshot/policy/policy_manager_test.go]: done

File: snapshot/policy/policy_manager.go [hunk 6]
```diff
@@ -288,3 +293,65 @@ func labelsForSource(si snapshot.SourceInfo) map[string]string {
...
+	// special case for root, return root path itself (either slash or backslash-separated)
+	if last == 0 {
+		return vol + pth[0:1]
+	}
+
+	return vol + pth[0:last]
+}
+
+// volumeAndPath splits path 'p' into Windows-specific volume (e.g. "X:" and path after that starting with either slash or backslash)
+func volumeAndPath(p string) (vol, path string) {
```
Comment (julio-lopez) [inline snapshot/policy/policy_manager.go:320]: Would https://golang.org/pkg/path/filepath/#VolumeName help simplify things here?

Comment (jkowalski) [reply snapshot/policy/policy_manager.go:320]: no, because it's platform-specific.

**Pull request merged**